### PR TITLE
RWA-1181-TaskSearchController-refactoring

### PR DESCRIPTION
**Pull request by serdaroguzhanakin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RWA-1181

### Change description ###
When accessControlResponse is empty TaskSearchController refactored to return 200 with an empty list

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

**Review by satvinder-hullait — DISMISSED** [review]

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/auth/access/AccessControlServiceTest.java [hunk 1]
```diff
@@ -114,4 +116,54 @@ void get_roles_given_user_id_should_return_empty_list() {
...
+        Optional<AccessControlResponse> result = accessControlService.getAccessControlResponse(idamToken);
+
+        assertTrue(result.isEmpty());
+
+        verify(idamService, times(1)).getUserInfo(idamToken);
+        verifyNoMoreInteractions(idamService);
+
+        verify(roleAssignmentService, times(1)).getRolesForUser(mockedUserInfo.getUid(), idamToken);
+        verifyNoMoreInteractions(roleAssignmentService);
+    }
```
Comment (raghera) [inline src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/auth/access/AccessControlServiceTest.java:167]: Maybe you can add one test when the service receives a `NoRoleAssignmentsFoundException` exception?  The result should still be an optional.empty().

**Review by raghera — COMMENTED** [review]
Looks really good Oguz.  Just one comment to add another unit test but otherwise all good.  👍

File: src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/auth/access/AccessControlServiceTest.java (hunk 1 above)
Comment (serdaroguzhanakin) [reply src/test/java/uk/gov/hmcts/reform/wataskmanagementapi/auth/access/AccessControlServiceTest.java:167]: `should_return_optional_empty_when_thrown_NoRoleAssignmentsFoundException` method was added

**Review by satvinder-hullait — APPROVED** [review]

**Review by mkayad — APPROVED** [review]
LGTM

**Pull request merged**